CARDS-2175: form-completion-status tests

sofi2002sofi opened the pull request.

sofi2002sofi committed:
CARDS-2175: Increase test coverage cards-form-completion-status

tests for AnswerCompletionStatusEditor

sofi2002sofi committed:
CARDS-2175: Increase test coverage cards-form-completion-status

tests for AnswerCompletionStatusEditorProvider

sofi2002sofi committed:
CARDS-2175: Increase test coverage cards-form-completion-status

tests for MinMaxAnswersValidator

sofi2002sofi committed:
CARDS-2175: Increase test coverage cards-form-completion-status

tests for DateRangeMinMaxAnswersValidator

sofi2002sofi committed:
CARDS-2175: Increase test coverage cards-form-completion-status

tests for MinMaxValueValidator

sofi2002sofi committed:
CARDS-2175: Increase test coverage cards-form-completion-status

tests for ConditionalSectionUtils

veronikaslc commented:
rebased on the latest `dev` and build, got the output:

 FAILURE! - in io.uhndata.cards.formcompletionstatus.AnswerCompletionStatusEditorTest
[ERROR] enterForEmptyFormNode(io.uhndata.cards.formcompletionstatus.AnswerCompletionStatusEditorTest)  Time elapsed: 0.338 s  <<< FAILURE!
java.lang.AssertionError
        at io.uhndata.cards.formcompletionstatus.AnswerCompletionStatusEditorTest.enterForEmptyFormNode(AnswerCompletionStatusEditorTest.java:163)

[ERROR] enterForFormNodeAddsDifferentStatusFlags(io.uhndata.cards.formcompletionstatus.AnswerCompletionStatusEditorTest)  Time elapsed: 0.373 s  <<< FAILURE!
java.lang.AssertionError
        at io.uhndata.cards.formcompletionstatus.AnswerCompletionStatusEditorTest.enterForFormNodeAddsDifferentStatusFlags(AnswerCompletionStatusEditorTest.java:148)

sofi2002sofi committed:
fixed errors caused by rebase

veronikaslc committed:
CARDS-2175: Increase test coverage cards-form-completion-status
Added missing pom dependencies
Removed unused initialAnswer parameter from validate() method

sofi2002sofi committed:
CARDS-2175: Increase test coverage of cards-form-completion-status

sdumitriu committed:
CARDS-2175: Increase test coverage of cards-form-completion-status
Modernize the revived tests and complete the module coverage to 100%:
- migrate the tests to the current platform APIs: jakarta.json, Mockito 5,
  sling-mock 4
- drop the Mockito runner in favor of plain mocks with reflection
  injection, since concurrently running runners collide on Mockito's
  global listener registry under the parallel surefire configuration
- raise the sling-mock resource resolver factory timeout, the default
  500ms is not always enough when several Oak-backed test contexts start
  in parallel
- add tests for the previously uncovered classes: RegExpValueValidator,
  RangeMinMaxAnswersValidator, and the AnswerValidator default methods
- cover the previously untested paths: the includes/excludes comparator
  family, empty and multivalued operands, unknown comparators, nested
  answers and questions, broken conditional definitions and values,
  answer options and suggested-only limits in MinMaxValueValidator,
  submitted forms and conditional sections in the completion status
  editor
- extend the date fallback in conditional values: fixed dates that are
  not in the strict JCR format are now parsed with the lenient supported
  formats instead of failing the comparison with a ClassCastException,
  in the spirit of the CARDS-2877 fix
- remove dead code: the never-invoked debugging toString methods, an
  unreachable summarize catch block, and a stream reduce combiner that
  can never run on a sequential stream; guard against sections that
  belong to no questionnaire
- rename the remaining vague test names and drop the meaningless
  constructor smoke test
- remove the module's coverage exemption, restoring the default 1.00
  required instruction coverage ratio
